### PR TITLE
Fix the custom vnet test template to provide the route table and nsg names

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What type of PR is this?**
/kind failing-test

**What this PR does / why we need it**: This is fixing the flake observed in https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-azure/1994/pull-cluster-api-provider-azure-e2e/1486169658842157056. We were creating a custom vnet with subnets and route table + nsg out of band, but not setting the route table and NSG name in the template. This caused CAPZ to skip reconciliation for route tables and NSGs, but then to pass default RT and NSG names to the cloud provider, and cloud provider would create the route table with default name to use it instead of the one we created. The flake was caused by the fact that the extra route table didn't get properly deleted which caused the cleanup step to fail.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by CecileRobertMichon** [comment]
/retest

same flake as #1995 , there might be something going on with AzureBastions

File: test/e2e/azure_test.go [hunk 1]
```diff
@@ -141,7 +142,7 @@ var _ = Describe("Workload cluster creation", func() {
 
 	if os.Getenv("LOCAL_ONLY") != "true" {
 		Context("Creating a private cluster", func() {
-			It("Creates a public management cluster in the same vnet", func() {
+			It("Creates a public management cluster a custom vnet", func() {
```
Comment (jackfrancis) [inline test/e2e/azure_test.go:145]: nit: "in a custom vnet"

If this test run passes lgtm and merge :)

**Review by jackfrancis — APPROVED** [review]
lgtm

File: test/e2e/azure_test.go (hunk 1 above)
Comment (CecileRobertMichon) [reply test/e2e/azure_test.go]: fixed!

**Review by jackfrancis — APPROVED** [review]
/lgtm

**Pull request merged**